File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String b, int k) {
        int wc=0;
        int minw=Integer.MAX_VALUE;
        for(int i=0;i<k;i++){
            if(b.charAt(i)=='W') wc++;
        }
        minw=wc;
        for(int i=k;i<b.length();i++){
            if(b.charAt(i)=='W') wc++;
            if(b.charAt(i-k)=='W') wc--;
            minw=Math.min(wc,minw);
        }
        return minw;
    }
}

File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.cpp
class Solution {
public:
    int minimumRecolors(string b, int k) {
        int m=INT_MAX;
        int wc=0;
        for(int i=0;i<k;i++){
            if(b[i]=='W') wc++;
        }
        m=wc;
        for(int i=k;i<b.length();i++){
            if(b[i]=='W') wc++;
            if(b[i-k]=='W') wc--;
            m=min(m,wc);
        }
        return m;
    }
};